File: Scripts/Player/PlayerStateMachine/PlayerStates/PlayerIdle.cs
using UnityEngine;

public class PlayerIdle : IPlayerState
{
    /// <summary>Determines if player state needs to change states and calls Idle method</summary>
    /// <param name="player">Instance of player class</param>
    /// <returns>Returns IPlayerState interface</returns>
    public IPlayerState DoState(Player player)
    {
        if (Input.GetKey(KeyCode.Space) && player.nextDodge < Time.time)
            return player.playerDodge;
        if (Input.GetAxisRaw("Horizontal") != 0 || Input.GetAxisRaw("Vertical") != 0)
            return player.playerMove;
        return player.playerIdle;
    }

    public void Idle()
    {
        // ToDo: Play animation that corresponds to look direction
    }
}

File: Scripts/Player/PlayerStateMachine/PlayerStates/player_idle.py
import time

import pygame

from Scripts.Player.PlayerStateMachine.player_state import PlayerState


class PlayerIdle(PlayerState):


    def __init__(self):
        super().__init__(None)


    def do_state(player):
        """
        Determines if player state needs to change states and calls Idle method
        :param player: Instance of player class
        :returns: Returns IPlayerState interface
        """

        # Get the state of all keys
        keys = pygame.key.get_pressed()

        # Update player position based on key input
        if any([keys[pygame.K_w], keys[pygame.K_s], keys[pygame.K_a], keys[pygame.K_d]]):
            return player.playerMove

        if keys[pygame.K_SPACE] and player.next_dodge < time.time():
            return player.player_dodge

        return player.playerIdle

    def idle(self):
        # ToDo: Play animation that corresponds to look direction
        raise NotImplementedError
